src/SmartComponents/FormComponents/BulkSelectButton.js

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useRef } from 'react';
+import { Button } from '@patternfly/react-core';
+import PropTypes from 'prop-types';
+import useFormApi from '@data-driven-forms/react-form-renderer/use-form-api';
+import useFieldApi from '@data-driven-forms/react-form-renderer/use-field-api';
+import { BULK_SELECT_BUTTON } from '../../Utilities/constants';
+import './BulkSelectButton.scss';
+
+const BulkSelectButton = (props) => {
+  const formOptions = useFormApi();
+  const dirtyFieldsRef = useRef(formOptions.getState().dirtyFields);
+  const {
+    group,
+    section,
+    input: { ...input },
+  } = useFieldApi({
+    ...props,
+    type: 'button',
+  });
+  useEffect(() => {
+    if (dirtyFieldsRef !== formOptions.getState().dirtyFields) {
```

Comment:
comparing arrays like that will not work https://www.freecodecamp.org/news/how-to-compare-arrays-in-javascript/
or you can use https://lodash.com/docs/4.17.15#isEqual

also for the ref, you have to call dirtyFieldsRef.current to actually access the value, the same for setting the value

Reply:
sorry im used to python, I'll change it right now

Reply:
Turns out you cannot use .current to set the value, only access it or else it will turn it to undefined.

Reply:
it should be possible to change it, you can take a look for examples here https://react.dev/reference/react/useRef